src/Files.App/ViewModels/SidebarViewModel.cs

```diff
@@ -240,91 +143,110 @@ public SidebarViewModel()
 			App.FileTagsManager.DataChanged += Manager_DataChanged;
 		}
 
-		private async void CreateItemHome()
+		public void UpdateSidebarSelectedItemFromArgs(string arg)
 		{
-			await CreateSection(SectionType.Home);
+			INavigationControlItem? item = null;
+			var sidebarItems = SideBarItems
+				.Where(x => !string.IsNullOrWhiteSpace(x.Path))
+				.Concat(SideBarItems
+					.Where(x => (x as LocationItem)?.ChildItems is not null)
+					.SelectMany(x => ((LocationItem)x).ChildItems)
+					.Where(x => !string.IsNullOrWhiteSpace(x.Path))
+				)
+				.ToList();
+
+			if (string.IsNullOrEmpty(arg))
+				return;
```

Comment:
I think it would make more sense if these two lines were moved to the beginning of the method